File: backend/go/internal/domain/result.go
package domain

import "time"

type Result struct {
	Username      string     `bson:"username" binding:"required"`
	QuestionCount int        `bson:"questionCount" binding:"required"`
	Result        float64    `bson:"result" binding:"required"`
	Date          *time.Time `bson:"date" binding:"required"`
}

File: backend/nodejs/result.js
const {Schema, model} = require('mongoose');

const Result = new Schema({
    username: {type: String, required: true},
    questionCount: {type: Number, required: true},
    result: {type: Number, required: true},
    date: {type: Date, required: true}
});

module.exports = model('Result', Result);
